src/main/java/ca/uhn/fhir/jpa/starter/CustomAuthorizationInterceptor.java

```diff
@@ -112,6 +114,23 @@
 		logger.info("Authentication failure");
 		return denyAll();
 	}
+	
+	private List<IAuthRule> allowForClaimResourceId(RequestDetails theRequestDetails,String patientId) {
+		return new RuleBuilder().allow().read().allResources()
```

Comment:
The formatting of this makes it a little confusing. The flow / pseudo-code is:
```
  .allowReadsInPatientCompartment().andThen()
  .allowWritesInPatientCompartment().andThen()
  .denyEverythingElse()
```

 Please change the code to:
```
return new RuleBuilder()
  .allow().read().allResources().inCompartment("Patient", new IdType("Patient", patientId)).andThen()
  .allow().write().allResources().inCompartment("Patient", new IdType("Patient", patientId)).andThen()
  .denyAll()
  .build();
```